### PR TITLE
Floating point support

### PR DESCRIPTION
**Pull request by enjhnsn2**
I wrote some high level bindings for floating point arithmetic. These include the necessary bindings to create constants, perform simple arithmetic (+, -, *, /, unary negative), cast to/from Dynamic ASTs, and some convenience functions for working specifically with float32s and doubles. Let me know what you think.

### CODE REVIEW

File: z3/src/ast.rs [hunk 1]
```diff
@@ -451,6 +465,18 @@ impl<'ctx> Real<'ctx> {
     }
 }
 
+impl<'ctx> Float<'ctx> {
+    pub fn from_f32(ctx: &'ctx Context, value: f32) -> Float<'ctx> {
+        let sort = Sort::float32(ctx);
+        Self::new(ctx, unsafe{Z3_mk_fpa_numeral_float(ctx.z3_ctx, value, sort.z3_sort)})
```
Comment (fitzgen) [inline z3/src/ast.rs:471]: Can you `cargo fmt` this?

File: z3/src/ast.rs [hunk 2]
```diff
@@ -451,6 +465,18 @@ impl<'ctx> Real<'ctx> {
...
+    }
+
+    pub fn from_f64(ctx: &'ctx Context, value: f64) -> Float<'ctx> {
```
Comment (fitzgen) [inline z3/src/ast.rs:474]: Can you also add a doc comment for this method and also `from_f32`?

**Review by fitzgen — COMMENTED** [review]
Thanks! This is pretty much good to merge, but I have a couple tiny comments below. Biggest thing is I'd like to have documentation comments for all the new methods. I want to make it so that, over time, we can make this crate's documentation coverage better than the current fairly-undocumented current state.

**Review by fitzgen — APPROVED** [review]
Thanks!

**Comment by fitzgen** [comment]
Sorry, I merged another PR since you pushed this one, would you mind rebasing to resolve the conflicts?

**Comment by enjhnsn2** [comment]
I just rebased with the new changes. Assuming the checks pass, we should be good to go :)

**Pull request merged**